staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -260,7 +296,21 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 	// This is safe as long as there is at most one call to processEvent in flight
 	// at any point in time.
 	if w.eventHandler != nil {
-		w.eventHandler(watchCacheEvent)
+		// We copy the event to avoid storing serializations in-memory for a longer
+		// time period. If we could tollerate higher memory usage, we can get rid of it.
+		wce := &watchCacheEvent{
```

Comment:
can we shallow copy this so we don't drift if new fields are added?

Reply:
and add a test (for various watch event types) that ensures setCachingObjects will not mess with anything beyond the top-level fields, so a shallow copy is sufficient

Reply:
done